components/timeout/timeout.js

```diff
@@ -0,0 +1,138 @@
+import domready from '../../assets/js/domready'
+import dialog from '../../assets/js/dialog'
+import fetch from '../../assets/js/fetch'
+import LoaderBtn from '../../assets/js/loader-btn'
+import { padStart } from 'lodash'
+
+class TimeoutUI {
+  constructor(timePrompt, timeLimit) {
+    this.el = document.querySelector('.js-timeout')
+    this.circleEl = document.querySelector('.js-timeout-circle')
+    this.timeTextEl = document.querySelector('.js-timeout-time')
+    this.timePrompt = timePrompt
+    this.timeLimit = timeLimit
+
+    if (!this.el || !this.circleEl || !this.timeTextEl) return false
+
+    this.circleElRadius = this.circleEl.getAttribute('r')
+    this.strokeWidth = this.circleEl.getAttribute('stroke-width')
+    this.timeStartCountdown = this.getTimeNow()
+
+    this.reset()
+  }
+
+  getTimeNow() {
+    return parseInt(new Date().getTime() / 1000)
+  }
+
+  onTick() {
+    let countDown = this.timeLimit - (this.getTimeNow() - this.timeStartCountdown)
+    if (countDown <= this.timePrompt) {
+      this.draw(countDown)
+    }
+
+    return countDown
+  }
+
+  draw(time) {
+    const date = new Date(null)
+    date.setSeconds(time)
+    const mins = padStart(date.getUTCMinutes(), 2, '0')
+    const seconds = padStart(date.getUTCSeconds(), 2, '0')
+    const angle = (360 - (360 / (this.timePrompt / time))) / (57.5 / this.circleElRadius)
+
+    this.timeTextEl.innerHTML = `${mins}:${seconds}`
+
+    if (angle > 0) {
+      // adjust for stroke, but add a couple of px to account for stroke radius
+      this.circleEl.style.strokeDasharray = `${Math.max(0, (angle - (this.strokeWidth / 2)) + 2)}, 1000`
+      this.circleEl.getBoundingClientRect()
+    }
+
+    if (time < 10) {
+      this.el.classList.add('is-warning')
+    }
+  }
+
+  reset() {
+    this.timeStartCountdown = this.getTimeNow()
+    this.circleEl.style.strokeDasharray = '0, 1000'
+    this.circleEl.getBoundingClientRect()
+  }
+}
+
+domready(() => {
+  const sessionTimeout = window.__EQ_SESSION_TIMEOUT__
```

Comment:
Should this session timeout const should be made product agnostic because the component is included in the `components.js` file and available to any system calling the pattern library?